src/js/events/CosmosPackagesActions.js

```diff
@@ -382,68 +384,34 @@ const CosmosPackagesActions = {
   },
 
   fetchRepositories(type) {
-    RequestUtil.json({
-      contentType: getContentType({
-        action: "repository.list",
-        actionType: "request",
-        entity: "package",
-        version: "v1"
-      }),
-      headers: {
-        Accept: getContentType({
-          action: "repository.list",
-          actionType: "response",
-          entity: "package",
-          version: "v1"
-        })
-      },
-      method: "POST",
-      url: `${Config.rootUrl}${Config.cosmosAPIPrefix}/repository/list`,
-      data: JSON.stringify({ type }),
-      success(response) {
+    repositoriesStream.fetchRepositories(type).subscribe(
+      function success(response) {
         AppDispatcher.handleServerAction({
           type: REQUEST_COSMOS_REPOSITORIES_LIST_SUCCESS,
           data: response.repositories
         });
       },
-      error(xhr) {
+      function error({ xhr = {} }) {
         AppDispatcher.handleServerAction({
           type: REQUEST_COSMOS_REPOSITORIES_LIST_ERROR,
           data: RequestUtil.getErrorFromXHR(xhr),
           xhr
         });
       }
-    });
+    );
   },
 
   addRepository(name, uri, index) {
-    RequestUtil.json({
-      contentType: getContentType({
-        action: "repository.add",
-        actionType: "request",
-        entity: "package",
-        version: "v1"
-      }),
-      headers: {
-        Accept: getContentType({
-          action: "repository.add",
-          actionType: "response",
-          entity: "package",
-          version: "v1"
-        })
-      },
-      method: "POST",
-      url: `${Config.rootUrl}${Config.cosmosAPIPrefix}/repository/add`,
-      data: JSON.stringify({ name, uri, index }),
-      success(response) {
+    repositoriesStream.addRepository(name, uri, index).subscribe(
+      function success(response) {
         AppDispatcher.handleServerAction({
           type: REQUEST_COSMOS_REPOSITORY_ADD_SUCCESS,
           data: response,
           name,
           uri
         });
       },
-      error(xhr) {
+      function error({ xhr = {} }) {
```

Comment:
I still think that with us removing the error messages this PR should be against `features/data-layer` and not master. The reason being when it touches master we are removing functionality because we will lose the error messages that the http-service does not provide.

by the way, do we have a ticket for that?

Reply:
https://jira.mesosphere.com/browse/DCOS-21727

Reply:
Since this PR has a legit fail of integration tests I'm inclining following your advice here and change the base.

Reply:
👍